pkg/frontend/validate.go

```diff
@@ -189,11 +189,19 @@ func validateNetworkInterfaceName(nicName string) error {
 	return nil
 }
 
-func validateAdminVMSize(vmSize string) error {
+func validateAdminMasterVMSize(vmSize string) error {
 	if vmSize == "" {
 		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "", "The provided vmSize '%s' is invalid.", vmSize)
 	}
-	return nil
+
+	// check to ensure that the size is supported as a master size
+	for k := range validate.SupportedMasterVmSizes {
+		if string(k) == vmSize {
```

Comment:
since `vmSize` is coming off the URL I believe it will be [lowercased](https://github.com/Azure/ARO-RP/blob/master/pkg/frontend/middleware/lowercase.go), should this be `strings.EqualFold()` instead?

Reply:
Good catch, I've changed the comparison and added tests